modular_skyrat/modules/drone_adjustments/drone.dm

```diff
@@ -0,0 +1,61 @@
+/obj/machinery/drone_dispenser/Initialize(mapload)
+	//So that there are starting drone shells in the beginning of the shift
+	if(mapload)
+		starting_amount = 10000
+	return ..()
+
+/obj/item/card/id/advanced/simple_bot
+	//So that the drones can actually access everywhere and fix it
+	trim = /datum/id_trim/centcom
+
+//This is so we log all machinery interactions for drones
+/obj/machinery/attack_drone(mob/living/simple_animal/drone/user, list/modifiers)
+	. = ..()
+	user.log_message("[key_name(user)] interacted with [src] at [AREACOORD(src)]", LOG_GAME)
+
+/obj/machinery/door
+	///The types of things that can slide under from bumping
+	var/static/list/sliding_under_types = list(
+		/mob/living/simple_animal/drone,
+		/mob/living/simple_animal/cortical_borer,
+	)
+
+//This allows drones to slide under/through a door
+/obj/machinery/door/Bumped(atom/movable/movable_atom)
+	. = ..()
+	addtimer(CALLBACK(src, .proc/sliding_under, movable_atom), 5)
```

Comment:
I don't know why, but I got a feeling that this would make sense as a component, but we both know I'm not the most knowledgeable at components.